templates/default/mesos_master.yaml.erb

```diff
@@ -0,0 +1,13 @@
+# Generated by Chef, local modifications will be overwritten
+
+init_config:
+  <% if @init_config.key?('default_timeout') -%>
+  default_timeout: <%= @init_config['default_timeout'] %>
+  <% end -%>
+
+instances:
+  <% if @init_config.key?('instances') -%>
```

Comment:
The condition here doesn't look correct (if there's something to check it should be on the `@instances` variable directly, `@init_config` should never have an `instances` key).

More generally, what you could instead here would be to format the `@instances` array to yaml (and, if no `@instances` have been passed, i.e. when `@instances` is an empty array, just define an instance with default values). It would also allow users to set the other configuration options (for example `disable_ssl_validation` for `mesos_master`, see https://github.com/DataDog/dd-agent/blob/5.10.1/conf.d/mesos_master.yaml.example#L15).

So something like: (haven't checked that the syntax is correct but just to describe the logic I'm thinking of)

```erb
instances:
<% if @instances.empty? -%>
- url: "http://<%= @node['ipaddress'] %>:5050"
<% else %>
<%= JSON.parse((@instances).to_json).to_yaml %>
<% end -%>
```

This also applies to `mesos_slave` ;)